### PR TITLE
f/wpc-600 preview link

### PR DESCRIPTION
**Pull request by camilleory**
Preview Button added to article and pages editor, take the user the a temporary preview of the article or page (link valid for 1 hour)
Jan-13-2022 13-59-34

### CODE REVIEW

File: packages/editor/src/client/routes/articleEditor.tsx [hunk 1]
```diff
@@ -482,6 +497,18 @@ export function ArticleEditor({id}: ArticleEditorProps) {
                   )}
                 </div>
               }
+              rightChildren={
+                <IconButtonLink
+                  disabled={hasChanged || !id}
```
[the remark this answers is not in the document]
Comment (camilleory) [reply packages/editor/src/client/routes/articleEditor.tsx:502]: I'd like to write a new ticket for that if that's ok

[the remark this answers is not in the document]
Comment (camilleory) [reply packages/editor/src/client/routes/articleEditor.tsx:502]: I'd like to write a new ticket for that if that's ok

File: packages/editor/src/client/routes/articleEditor.tsx [hunk 2]
```diff
@@ -482,6 +497,18 @@ export function ArticleEditor({id}: ArticleEditorProps) {
...
+                  style={{marginTop: '4px'}}
+                  size={'lg'}
+                  icon={<Icon icon="eye" />}
+                  onClick={e => {
+                    window.open(data?.articlePreviewLink || '')
```
[the remark this answers is not in the document]
Comment (camilleory) [reply packages/editor/src/client/routes/articleEditor.tsx:507]: I get this error if I do: `Invalid hook call. Hooks can only be called inside of the body of a function component.`
Should I find a way around it or is it ok to leave the query outside ?

File: packages/editor/src/client/routes/articleEditor.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (roosnic1) [reply packages/editor/src/client/routes/articleEditor.tsx:502]: @camilleory 👍🏾

File: packages/editor/src/client/routes/articleEditor.tsx [hunk 3]
```diff
@@ -482,6 +491,19 @@ export function ArticleEditor({id}: ArticleEditorProps) {
...
                 </div>
               }
+              rightChildren={
+                <IconButtonLink
+                  disabled={hasChanged || !id}
+                  style={{marginTop: '4px'}}
+                  size={'lg'}
+                  icon={<Icon icon="eye" />}
+                  onClick={async e => {
+                    await previewLinkRefetch({id: id, hours: 1})
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleEditor.tsx:501]: `previewLinkRefetch()` returns the data here and you should use that link to open a new window or not.

File: packages/editor/src/client/routes/articleEditor.tsx [hunk 4]
```diff
@@ -482,6 +491,19 @@ export function ArticleEditor({id}: ArticleEditorProps) {
...
+                    window.open(data?.articlePreviewLink || '')
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleEditor.tsx:502]: Also if `previewLinkRefetch()` doesn't return anything you shouldn't open a new tab. Here you can just do nothing if the query doesn't return a link,

**Review by roosnic1 — CHANGES_REQUESTED** [review]
Much better. Just some tiny changes.

**Review by penk85 — DISMISSED** [review]

**Review by roosnic1 — APPROVED** [review]
👍🏾

**Review by bishoymelek — APPROVED** [review]
🎉

**Review by penk85 — APPROVED** [review]

**Pull request merged**